wurst/util/StringUtils.wurst

```diff
@@ -157,6 +126,101 @@ public function string.reduce(string reduced) returns string
 public function string.reduce() returns string
 	return this.reduce(" ")
 
+public interface ForStringSplitCallback
+	function callback(string str)
+
+/** Custom Function which can split strings and runs callback for each string.
+	It can run even with no delimiter present (with single element).
+	Example usage:
+	"H001,H002".forEachIn(",") ->
+		doSomething()  */
+public function string.forEachIn(string delim, ForStringSplitCallback cb)
+	if this.length() > 0
+		int i = 0
+		int pos = this.indexOf(delim, i)
+
+		if pos == -1
+			cb.callback(this.substring(i, this.length()))
+
+		while pos != -1
+			cb.callback(this.substring(i, pos))
+			pos += 1
+			i = pos
+			pos = this.indexOf(delim, pos)
+			if pos == -1
+				cb.callback(this.substring(i, this.length()))
+
+/** Creates list of IDs converted from Int to their String value delimited by comma (using toRawCode).*/
+public function createDelimitedString(vararg int ids) returns string
+	string s = ""
+	var paramCount = 0
+	for id in ids
+		paramCount++
+		s += id.toRawCode()
+		s += ","
+	if s != ""
+		s = s.substring(0, s.length()-1)
+	if paramCount > WC3LIMIT_FUNCTIONS_PARAMETER_COUNT
+		error(WC3LIMIT_FUNCTIONS_PARAMETER_COUNT_ERROR)
+	return s
+
+/** Creates one long String from all Strings delimited by comma.*/
+public function createDelimitedString(vararg string ids) returns string
+	string s = ""
+	var paramCount = 0
+	for id in ids
+		paramCount++
+		if id != ""
+			s += id
+			s += ","
+	if s != ""
+		s = s.substring(0, s.length()-1)
+	if paramCount > WC3LIMIT_FUNCTIONS_PARAMETER_COUNT
+		error(WC3LIMIT_FUNCTIONS_PARAMETER_COUNT_ERROR)
+	return s
+
+/** Checks whether a string is representing a digit or not.*/
+public function string.isDigit() returns bool
+	bool isDigit = false
+	for int i = 0 to this.length()-1
+		if this.charAt(i) == "0" or this.charAt(i) == "1" or this.charAt(i) == "2" or this.charAt(i) == "3" or this.charAt(i) == "4"
```

Comment:
Saving the return of this.charAt in a variable would save a lot of useless calls here.  
Also, there is a simpler implementation to this consisting of calling S2I to convert the string to number and checking if it's different from 0. If the conversion returns 0, then you have to check if the string is actually "0", because any string that can't be converted to integer returns 0. Obvisouly this method requires some experimentation to see what happens/should happen when the string contains the minus operator or has spaces in it.

Reply:
That is not entirely true.
If you have string like "241a14", the S2I function will actually convert this to 241. This function checks whether the full string can be considered a number, not just a part of it.
Your other suggestions are fine, I'll give it a look.